pkg/compiler/types.go

```diff
@@ -645,6 +645,16 @@ var typeString = &typeDesc{
 				NoZ:        t.Ident == stringnoz,
 			}
 		}
+		if len(args) > 0 && t.Ident == "dirname" {
```

Comment:
Please add some tests to pkg/compiler/testdata/{all,errors}.txt
I think this condition is wrong. It generates BufferDirname only if dirname has args. But I say just `dirname` without args, it should produce an error rather than silently generating normal string type.
This can be tested with pkg/compiler/testdata/errors*.txt